ctapipe/io/tableloader.py

```diff
@@ -166,12 +166,12 @@ def read_telescope_events_for_id(self, tel_id):
                     join_type="outer",
                 )
 
-        if self.shower_table is not None:
+        if self.shower_table is not None and len(table > 0)):
             table = join(
                 table, self.shower_table, keys=["obs_id", "event_id"], join_type="inner"
             )
 
-        if self.load_trigger and len(table) > 0:
+        if self.load_trigger:
```

Comment:
Why remove the length check? This errors when `len(table) == 0`, astropy cannot join empty tables which is the  reason the check was there

Reply:
reverted